pyzebra/fit2.py

```diff
@@ -0,0 +1,151 @@
+from lmfit import minimize, Parameters, Model
+from lmfit.models import LinearModel, LorentzianModel, GaussianModel
+import matplotlib.pyplot as plt
+from scipy.integrate import simps
+import scipy as sc
+from scipy import integrate
+import numpy as np
+from time import sleep
+
+
+def fitccl(data, keys, guess, vary, constraints_min, constraints_max, numfit_min=None, numfit_max=None):
+    """Made for fitting of ccl date where 1 peak is expected. Allows for combination of gaussian, lorentzian and linear model combination
+
+
+    :param data: dictionary after peak fining
+    :param keys: name of the measurement in the data dict (i.e. M123)
+    :param guess: initial guess for the fitting, if none, some values are added automatically in order (see below)
+    :param vary: True if parameter can vary during fitting, False if it to be fixed
+    :param numfit_min: minimal value on x axis for numerical integration - if none is centre of gaussian minus 3 sigma
+    :param numfit_max: maximal value on x axis for numerical integration - if none is centre of gaussian plus 3 sigma
+    :param constraints_min: min constranits value for fit
+    :param constraints_max: max constranits value for fit
+
+    :return data dict with additional values
+
+    order for guess, vary, constraints_min, constraints_max
+    [Gaussian centre, Gaussian sigma, Gaussian amplitude, Lorentzian centre, Lorentzian sigma, Lorentzian amplitude, background slope, background intercept]
+    examples:
+    guess = [None, None, 100, None, None, None, 0, None]
+    vary = [True, True, True, True, False, True, True,  True]
+    constraints_min = [23, None, 50, None, None, None, 0, 0]
+    constraints_min = [80, None, 1000, None, None, None, 0, 100]
+    """
+
+    if len(data["Measurements"][str(keys)]["peak_indexes"]) == 1:

```

Comment:
Same with the guard clause. For long functions it's better to do the early exit, rather than indent the entire function.

Reply:
done 👍